### PR TITLE
Compare 'nothing' using === not ==

### PR DESCRIPTION
**Pull request by fingolfin**
... for optimal code results. While in most cases it won't matter, it is
a good habit to just always adhere to this.

### CODE REVIEW

File: src/Combinatorics/Matroids/properties.jl [hunk 1]
```diff
@@ -161,7 +161,7 @@ function flats_impl(M::Matroid, r::Union{Int,Nothing}, num_flats::Int, pm_flats)
         jl_flats = reverse(jl_flats)
     end
     matroid_flats = [[M.groundset[i] for i in flat] for flat in jl_flats]
-    if r!=nothing
+    if r != nothing
```
Comment (thofma) [inline src/Combinatorics/Matroids/properties.jl:164]: something is missing here

File: src/Groups/matrices/MatGrp.jl [hunk 2]
```diff
@@ -341,9 +340,10 @@ function (G::MatrixGroup)(x::MatrixGroupElem; check::Bool=true)
    else
       _is_true, x_gap = lies_in(x.elm,G,nothing)
       @req _is_true "Element not in the group"
-      if x_gap==nothing return MatrixGroupElem(G,x.elm)
-      else return MatrixGroupElem(G,x.elm,x_gap)
+      if x_gap == nothing
```
Comment (thofma) [inline src/Groups/matrices/MatGrp.jl:343]: here too

**Comment by lkastner** [comment]
> ... for optimal code results. While in most cases it won't matter, it is a good habit to just always adhere to this.

Why not use `isnothing`? Sure, it does the same thing as `===`, but wouldn't this be the better style, to use the dedicated function?

**Comment by thofma** [comment]
https://docs.julialang.org/en/v1.6/manual/performance-tips/#Checking-for-equality-with-a-singleton

(Should be the same starting with 1.7)

**Review by HereAround — APPROVED** [review]
Thank you @fingolfin . Looks good to me.

**Comment by thofma** [comment]
There were a few cases missing, which should have been fixed before merging this

**Comment by lkastner** [comment]
I still don't understand why not to use `isnothing`. The singleton comparison statement did not enlighten me (I know that `Nothing` is a singleton), I am sorry.

**Comment by fingolfin** [comment]
I wanted to minimize the changes, that was very easy with a reggex changing `==nothing` to `===nothing`, would have been much harder to change it to `isnothing`. That said, I also don't see anything "better" about using `isnothing`, seems like a purely subjective matter of taste.

File: src/Combinatorics/Matroids/properties.jl (hunk 1 above)
Comment (fingolfin) [reply src/Combinatorics/Matroids/properties.jl:164]: Indeed *sigh*. I'll send a follow up PR.

**Comment by lkastner** [comment]
`isnothing` is the dedicated Julia function for figuring out if something is `Nothing`. So if at some point comparing to `Nothing` should be done in a different way than `===` this would automatically be changed for us if we used `isnothing`. Similar to how we are encouraged not to directly access members. But I guess this issue is really irrelevant, so I will stop here.

**Comment by HereAround** [comment]
(My apologies for moving this along too quickly.)

**Pull request merged**